cabal-install/Distribution/Client/Dependency/TopDown.hs

```diff
@@ -669,8 +671,9 @@ finaliseSelectedPackages pref selected constraints =
         -- version constraints. TODO: distinguish hacks from prefs
         bounded = boundedAbove versionRange
         isPreferred p
-          | bounded   = True -- any constant will do
-          | otherwise = packageVersion p `withinRange` preferredVersions
+          | bounded   = 999 -- any constant will do
```

Comment:
I am not sure here about this `bounded` case, what weight should we assign to it relative to the other preferences? Not sure I understand this bit.